median/plugin.go

```diff
@@ -65,3 +74,45 @@ func (r *reportingPluginFactoryService) Close() error {
 func (r *reportingPluginFactoryService) HealthReport() map[string]error {
 	return map[string]error{r.Name(): r.Healthy()}
 }
+
+// medianContract adapts a [types.ChainReader] to [median.MedianContract].
+type medianContract struct {
+	chainReader types.ChainReader
+	contract    types.BoundContract
+}
+
+type latestTransmissionDetailsResponse struct {
+	ConfigDigest    ocrtypes.ConfigDigest
+	Epoch           uint32
+	Round           uint8
+	LatestAnswer    *big.Int
+	LatestTimestamp time.Time
+}
+
+type latestRoundRequested struct {
+	ConfigDigest ocrtypes.ConfigDigest
+	Epoch        uint32
+	Round        uint8
+}
+
+func (m *medianContract) LatestTransmissionDetails(ctx context.Context) (configDigest ocrtypes.ConfigDigest, epoch uint32, round uint8, latestAnswer *big.Int, latestTimestamp time.Time, err error) {
+	var resp latestTransmissionDetailsResponse
+
+	err = m.chainReader.GetLatestValue(ctx, m.contract, "LatestTransmissionDetails", nil, &resp)
+	if err != nil {
+		return
+	}
+
+	return resp.ConfigDigest, resp.Epoch, resp.Round, resp.LatestAnswer, resp.LatestTimestamp, nil
+}
+
+func (m *medianContract) LatestRoundRequested(ctx context.Context, lookback time.Duration) (configDigest ocrtypes.ConfigDigest, epoch uint32, round uint8, err error) {
+	var resp latestRoundRequested
+
+	err = m.chainReader.GetLatestValue(ctx, m.contract, "LatestRoundReported", map[string]any{"lookback": lookback}, &resp)
+	if err != nil {
+		return
+	}
+
+	return resp.ConfigDigest, resp.Epoch, resp.Round, nil
+}
```

Comment:
Actually, come to think of it, if we're going to do it this way then the definition of `medianContract` along with its methods needs to go in `chainlink-commons` rather than here.  Otherwise there is no way for `chainlink-commons` to import it.

Reply:
Why would chainlink-common need to import this? We only want to use it at the outermost client layer. If we used it on the server side, then we would still be using the GRPC services over the wire, and could not phase them out.

Reply:
Yeah, nevermind I think I was confused about the interconnection of all of this. I was thinking the `NewMedianFactory` method in `chainlink-common` was calling `NumericalMedianFactory` directly but I guess the call it's making must end up getting back to this function (in `chainlink-feeds` at some point, where the actual call gets made... right?

Reply:
Kind of yeah. I think it's more about the GRPC layer being a separate concern from chain reader being available or not.